contracts/tools/DepositHelper.sol

```diff
@@ -0,0 +1,125 @@
+  mapping(address => bool) whitelist;
+
+  modifier onlyOneCallPerBlockPerVault(address vault_) {
+    require(calls[msg.sender][vault_] < block.number, "DH: call in the same block forbidden");
+    _;
+    calls[msg.sender][vault_] = block.number;
+  }
+
+  // ******************** USERS ACTIONS *********************
+
+  /// @notice Deposits to vault specified underlying token amount. Caller receives share tokens.
+  /// @dev Approval for share token is assumed.
+  /// @param vault_ A target vault for deposit
+  /// @param underlyingAmount_ Amount of vault's underlying token for deposit
+  function depositToVault(address vault_, uint256 underlyingAmount_)
+  external nonReentrant onlyOneCallPerBlockPerVault(vault_) {
+    require(underlyingAmount_ > 1, "DH: not enough amount");
+
+    address underlying = ISmartVault(vault_).underlying();
+    IERC20(underlying).safeTransferFrom(msg.sender, address(this), underlyingAmount_);
+
+    IERC20(underlying).safeApprove(vault_, 0);
+    IERC20(underlying).safeApprove(vault_, underlyingAmount_);
+    ISmartVault(vault_).depositAndInvest(underlyingAmount_);
+
+    uint256 shareBalance = IERC20(vault_).balanceOf(address(this));
+    require(shareBalance != 0, "DH: zero shareBalance");
+
+    IERC20(vault_).safeTransfer(msg.sender, shareBalance);
+
+    // send change (if any) back
+    uint256 underlyingBalance = IERC20(underlying).balanceOf(address(this));
+    if (underlyingBalance > 0) {
+      IERC20(underlying).safeTransfer(msg.sender, underlyingBalance);
+    }
+  }
+
+  /// @notice Withdraws from vault specified share amount. Caller receives underlying token.
+  /// @dev Approval for share token is assumed.
+  /// @param vault_ A target vault withdraw from
+  /// @param shareTokenAmount_ Amount of vault's share token to withdraw
+  function withdrawFromVault(address vault_, uint256 shareTokenAmount_)
+  external nonReentrant onlyOneCallPerBlockPerVault(vault_) {
+    require(shareTokenAmount_ != 0, "DH: zero amount");
+
+    IERC20(vault_).safeTransferFrom(msg.sender, address(this), shareTokenAmount_);
+
+    ISmartVault(vault_).withdraw(shareTokenAmount_);
+
+    address underlying = ISmartVault(vault_).underlying();
+    uint256 underlyingBalance = IERC20(underlying).balanceOf(address(this));
+    require(underlyingBalance != 0, "DH: zero underlying balance");
+
+    IERC20(underlying).safeTransfer(msg.sender, underlyingBalance);
+
+    _claimAndSendAllRewards(vault_);
+
+    // send change (if any) back
+    uint256 shareBalance = IERC20(vault_).balanceOf(address(this));
+    if (shareBalance > 0) {
+      IERC20(vault_).safeTransfer(msg.sender, shareBalance);
+    }
+  }
+
+  /// @notice Claims rewards from vault. Caller receives all reward tokens.
+  /// @dev Approval for share token is assumed.
+  /// @param vault_ A target vault to claim rewards from
+  function getAllRewards(address vault_)
+  external nonReentrant onlyOneCallPerBlockPerVault(vault_) {
```

Comment:
you can remove onlyOneCallPerBlockPerVault
claim can be called in the same block, it is not necessary to have a strict access